components/cam/src/physics/cam/micro_p3.F90

```diff
@@ -3045,6 +2888,425 @@ subroutine check_values(Qv,T,i,timestepcount,force_abort,source_ind)
+   else if (t .ge. zerodegc) then
+      ! rain number sink due to collection
+      ! for T > 273.15, assume collected rain number is shed as
+      ! 1 mm drops
+      ! note that melting of ice number is scaled to the loss
+      ! rate of ice mass due to melting
+      ! collection of rain above freezing does not impact total rain mass
+      nrcol  = 10._rtype**(f1pr07 + logn0r)*rho*rhofaci*eri*nitot_incld     
+      ! for now neglect shedding of ice collecting rain above freezing, since snow is
+      ! not expected to shed in these conditions (though more hevaily rimed ice would be
+      ! expected to lead to shedding)
+   end if 
+
+  end subroutine ice_drop_collection
+
+  subroutine ice_self_collection(rho,rhofaci,    &
+  f1pr03,eii,Eii_fact,qitot_incld,nitot_incld,    &
+             nislf)
+
+   ! self-collection of ice 
+
+   ! here we multiply rates by collection efficiency, air density,
+   ! and air density correction factor since these are not included
+   ! in the lookup table calculations
+   ! note 'f1pr' values are normalized, so we need to multiply by N
+
+   implicit none 
+
+   real(rtype), intent(in) :: rho 
+   real(rtype), intent(in) :: rhofaci
+   real(rtype), intent(in) :: f1pr03  
+   real(rtype), intent(in) :: eii 
+   real(rtype), intent(in) :: Eii_fact 
+   real(rtype), intent(in) :: qitot_incld
+   real(rtype), intent(in) :: nitot_incld
+
+   real(rtype), intent(out) :: nislf 
+
+   if (qitot_incld.ge.qsmall) then
+      nislf = f1pr03*rho*eii*Eii_fact*rhofaci*nitot_incld
+   endif
+
+
+end subroutine ice_self_collection
+
+
+subroutine ice_melting(rho,t,pres,rhofaci,    &
+f1pr05,f1pr14,xxlv,xlf,dv,sc,mu,kap,qv,qitot_incld,nitot_incld,    &
+           qimlt,nimlt)
+   ! melting
+   ! need to add back accelerated melting due to collection of ice mass by rain (pracsw1)
+   ! note 'f1pr' values are normalized, so we need to multiply by N
+   ! currently enhanced melting from collision is neglected
+   ! include RH dependence
+
+   implicit none 
+
+   real(rtype), intent(in) :: rho
+   real(rtype), intent(in) :: t 
+   real(rtype), intent(in) :: pres 
+   real(rtype), intent(in) :: rhofaci 
+   real(rtype), intent(in) :: f1pr05 
+   real(rtype), intent(in) :: f1pr14 
+   real(rtype), intent(in) :: xxlv 
+   real(rtype), intent(in) :: xlf
+   real(rtype), intent(in) :: dv 
+   real(rtype), intent(in) :: sc 
+   real(rtype), intent(in) :: mu 
+   real(rtype), intent(in) :: kap
+   real(rtype), intent(in) :: qv 
+   real(rtype), intent(in) :: qitot_incld 
+   real(rtype), intent(in) :: nitot_incld 
+
+   real(rtype), intent(out) :: qimlt 
+   real(rtype), intent(out) :: nimlt
+
+   real(rtype) :: qsat0
+
+   if (qitot_incld .ge.qsmall .and. t.gt.zerodegc) then
+      qsat0 = 0.622_rtype*e0/(pres-e0)
+
+      qimlt = ((f1pr05+f1pr14*sc**thrd*(rhofaci*rho/mu)**0.5_rtype)*((t-   &
+      zerodegc)*kap-rho*xxlv*dv*(qsat0-qv))*2._rtype*pi/xlf)*nitot_incld
+
+
+      qimlt = max(qimlt,0.)
```

Comment:
Does this mean that qimlt is always negative?  If so then how do rectify that it is later used as a "source" for rain, see 

```
! rain
          sinks   = (qrevp+qrcol+qrheti)*dt
          sources = qr(i,k) + (qrcon+qcaut+qcacc+qimlt+qcshd)*dt
```

This tells me that the presence of qimlt would lead to a smaller source term which wouldn't make sense.  Similarly qimlt is listed as a "sink" for ice.  Thus when it is applied to rain it will effectively remove rain and add ice which is the opposite of what we would expect.  See,

```
qr(i,k) = qr(i,k) + (-qrcol+qimlt-qrheti+            & ...
 qitot(i,k) = qitot(i,k) - (qisub+qimlt)*dt
```